src/native/corehost/apphost/apphost.windows.cpp

```diff
@@ -61,11 +61,12 @@ namespace
         if (pal::getenv(_X("DOTNET_DISABLE_GUI_ERRORS"), &gui_errors_disabled) && pal::xtoi(gui_errors_disabled.c_str()) == 1)
             return;
 
-        pal::string_t dialogMsg = _X("To run this application, you must install .NET.\n\n");
+        pal::string_t dialogMsg;
         pal::string_t url;
         const pal::string_t url_prefix = _X("  - ") DOTNET_CORE_APPLAUNCH_URL _X("?");
         if (error_code == StatusCode::CoreHostLibMissingFailure)
         {
+            dialogMsg = pal::string_t(_X("To run this application, you must install .NET Desktop Runtime ")) + _STRINGIFY(COMMON_HOST_PKG_VER) + _X(" (") + get_arch() + _X(").\n\n");
```

Comment:
Thanks for adding the architecture to the error. I think we should also add this to the missing framework error.
There are multiple places for this, but the most important ones are:
https://github.com/dotnet/runtime/blob/e6e7ef94b78bd73e84fc4c58138535d6e5b39911/src/native/corehost/fxr/fx_resolver.messages.cpp#L115
and
https://github.com/dotnet/runtime/blob/e6e7ef94b78bd73e84fc4c58138535d6e5b39911/src/native/corehost/fxr/fx_resolver.messages.cpp#L119

I would just add "(x86)" to the message after the framework name (or version).

It's a minor change, but I think this will be crucial for the ARM64 cases.